superset-frontend/spec/javascripts/components/TableSelector_spec.jsx

```diff
@@ -18,267 +18,275 @@
-        schema: 'main',
-        label: 'birth_names',
-        title: 'birth_names',
+    it('should change table value', async () => {
+      act(() => {
+        wrapper
+          .find('[name="select-table"]')
+          .first()
+          .props()
+          .onChange(selectedTable);
       });
-      expect(wrapper.state().tableName).toBe('birth_names');
+      await waitForComponentToPaint(wrapper);
+      wrapper.update();
+      expect(
+        wrapper.find('[name="select-table"]').first().props().value[0],
+      ).toEqual(selectedTable);
     });
 
-    it('should call onTableChange with schema from table object', () => {
-      wrapper.setProps({ schema: null });
-      wrapper.instance().changeTable({
-        value: 'my_table',
-        schema: 'other_schema',
-        label: 'other_schema.my_table',
-        title: 'other_schema.my_table',
+    it('should call onTableChange with schema from table object', async () => {
+      act(() => {
+        wrapper
+          .find('[name="select-table"]')
+          .first()
+          .props()
+          .onChange(selectedTable);
       });
-      expect(mockedProps.onTableChange.getCall(0).args[0]).toBe('my_table');
-      expect(mockedProps.onTableChange.getCall(0).args[1]).toBe('other_schema');
+      await waitForComponentToPaint(wrapper);
+      wrapper.update();
+      expect(mockedProps.onTableChange.getCall(0).args[0]).toBe('birth_names');
+      expect(mockedProps.onTableChange.getCall(0).args[1]).toBe('main');
     });
   });
 
-  it('changeSchema', () => {
-    sinon.stub(wrapper.instance(), 'fetchTables');
+  describe('getTableNamesBySubStr', () => {
+    beforeEach(() => {
+      wrapper.setProps({ schema: 'main' });
+    });
+    afterEach(fetchMock.resetHistory);
+    afterAll(fetchMock.reset);
 
-    wrapper.instance().changeSchema({ label: 'main', value: 'main' });
-    expect(wrapper.instance().fetchTables.callCount).toBe(1);
-    expect(mockedProps.onChange.callCount).toBe(1);
-    wrapper.instance().changeSchema();
-    expect(wrapper.instance().fetchTables.callCount).toBe(2);
-    expect(mockedProps.onChange.callCount).toBe(2);
+    it('should handle empty', () =>
+      act(() => {
+        wrapper
+          .find('[name="async-select-table"]')
+          .first()
+          .props()
+          .loadOptions()
+          .then(data => {
+            expect(data).toEqual({ options: [] });
+          });
+      }));
 
-    wrapper.instance().fetchTables.restore();
+    it('should handle table name', () => {
+      fetchMock.get(GET_TABLE_ENDPOINT, tables, {
+        overwriteRoutes: true,
+      });
+      act(() => {
+        wrapper
+          .find('[name="async-select-table"]')
+          .first()
+          .props()
+          .loadOptions()
+          .then(data => {
+            expect(fetchMock.calls(GET_TABLE_ENDPOINT)).toHaveLength(1);
+            expect(data).toEqual(tables);
+          });
+      });
+      return Promise.resolve();
```

Comment:
Same here, I think this test will always succeed since we're returning a resolved promise here.